### PR TITLE
Require Cors header values to be specified with double quotes.

### PR DESCRIPTION
**Pull request by stalar**
*Issue #, if available:* #1381

*Description of changes:*

Require Cors header values to have double quoting for start-api. If quotes are missing, fail with an exception.

Verified by running the template from #1381 both locally and deployed.

*Checklist:*

- [ ] Write Design Document ([Do I need to write a design document?](https://github.com/awslabs/aws-sam-cli/blob/develop/DEVELOPMENT_GUIDE.rst#design-document))
- [ ] Write unit tests
- [x] Write/update functional tests
- [x] Write/update integration tests
- [x] `make pr` passes
- [ ] Write documentation

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by TheSriram** [comment]
Have re-kicked the build, somehow it didnt start, will review this once the builds are okay.

File: samcli/commands/local/lib/sam_api_provider.py [hunk 1]
```diff
@@ -103,23 +103,58 @@ def extract_cors(self, cors_prop):
...
             cors = Cors(
-                allow_origin=cors_prop.get("AllowOrigin"),
-                allow_methods=allow_methods,
-                allow_headers=cors_prop.get("AllowHeaders"),
-                max_age=cors_prop.get("MaxAge"),
+                allow_origin=allow_origin, allow_methods=allow_methods, allow_headers=allow_headers, max_age=max_age
             )
         elif cors_prop and isinstance(cors_prop, string_types):
+            allow_origin = cors_prop
+            if allow_origin[0] != "'" or allow_origin[-1] != "'":
```
Comment (TheSriram) [inline samcli/commands/local/lib/sam_api_provider.py]: if this is a string, can we just do startswith or endswith instead?

File: samcli/commands/local/lib/sam_api_provider.py [hunk 2]
```diff
@@ -103,23 +103,58 @@ def extract_cors(self, cors_prop):
...
+        cors_dict : dict
+            Resource properties for Cors
+
+        Return
+        ------
+        A string with the extra quotes removed
+        """
+        prop = cors_dict.get(prop_name)
+        if prop:
+            if (not isinstance(prop, string_types)) or prop[0] != "'" or prop[-1] != "'":
```
Comment (TheSriram) [inline samcli/commands/local/lib/sam_api_provider.py]: Similarly here

File: samcli/commands/local/lib/sam_api_provider.py (hunk 1 above)
Comment (stalar) [reply samcli/commands/local/lib/sam_api_provider.py]: Will fix!

**Comment by jfuss** [comment]
@stalar Thanks for the PR. Things look good. I did add two additional unit tests to verify exceptions are thrown correctly. 

@TheSriram Can you take a peek at this? I think its good to go but wanted another approval since I added some tests.

**Pull request merged**